src/Payum/Action/Api/CreatePayByLinkTransactionAction.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Payum\Action\Api;
+
+use CommerceWeavers\SyliusTpayPlugin\Payum\Factory\Token\NotifyTokenFactoryInterface;
+use CommerceWeavers\SyliusTpayPlugin\Payum\Request\Api\CreateTransaction;
+use CommerceWeavers\SyliusTpayPlugin\Tpay\Factory\CreatePayByLinkPayloadFactoryInterface;
+use Payum\Core\GatewayAwareInterface;
+use Payum\Core\GatewayAwareTrait;
+use Payum\Core\Reply\HttpRedirect;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Webmozart\Assert\Assert;
+
+final class CreatePayByLinkTransactionAction extends AbstractCreateTransactionAction implements GatewayAwareInterface
+{
+    use GatewayAwareTrait;
+
+    public function __construct(
+        private readonly CreatePayByLinkPayloadFactoryInterface $createPayByLinkPayloadFactory,
+        private readonly NotifyTokenFactoryInterface $notifyTokenFactory,
+    ) {
+        parent::__construct();
+    }
+
+    /**
+     * @param CreateTransaction $request
+     */
+    public function execute($request): void
+    {
+        /** @var PaymentInterface $model */
+        $model = $request->getModel();
+        $token = $request->getToken();
```

Comment:
We should avoid requesting the token to only get the gateway name. It prevents us from resuing it in API. Check the `src/Payum/Action/Api/CreateBlikLevelZeroTransactionAction.php:36` for a reference;